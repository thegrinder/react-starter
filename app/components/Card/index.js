import styled from 'styled-components';
import { getColor, rem } from 'basic-styled-uikit';

const Card = styled.div`
  background: ${getColor('white')};
  border-radius: ${rem(6)};
  box-shadow: 0 1px 3px 0 ${getColor('gray300')},
    0 0 0 1px ${getColor('gray300')};
`;

export default Card;
