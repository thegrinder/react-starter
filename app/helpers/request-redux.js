export const initialRequestState = {
  error: null,
  initialLoad: true,
  loading: false,
};

export const createRequest = (namespace) => {
  const actionTypes = {
    FAILED: `${namespace}/FAILED`,
    FULFILLED: `${namespace}/FULFILLED`,
    LOADING: `${namespace}/LOADING`,
    RESET: `${namespace}/RESET`,
  };

  const actionCreators = {
    failed: error => ({ error, type: actionTypes.FAILED }),
    fulfilled: () => ({ type: actionTypes.FULFILLED }),
    loading: () => ({ type: actionTypes.LOADING }),
    reset: () => ({ type: actionTypes.RESET }),
  };

  const reducer = (state = initialRequestState, action) => {
    switch (action.type) {
      case actionTypes.LOADING:
        return {
          ...state,
          error: null,
          loading: true,
        };
      case actionTypes.FAILED:
        return {
          ...state,
          error: action.error,
        };
      case actionTypes.FULFILLED:
        return {
          ...state,
          initialLoad: false,
          loading: false,
        };
      case actionTypes.RESET:
        return initialRequestState;
      default:
        return state;
    }
  };

  return {
    actionCreators,
    actionTypes,
    reducer,
  };
};
