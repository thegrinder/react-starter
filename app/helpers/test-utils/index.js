import React from 'react';
import PropTypes from 'prop-types';
import { IntlProvider } from 'react-intl';
import { ThemeProvider } from 'styled-components';
import { render } from 'react-testing-library';

import theme from '../../theme/components';

const propTypes = {
  children: PropTypes.node.isRequired,
};

const Providers = ({ children }) => (
  <IntlProvider locale="en">
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>
  </IntlProvider>
);

Providers.propTypes = propTypes;

const customRender = (ui, options) => render(ui, { wrapper: Providers, ...options });

export * from 'react-testing-library';
export { customRender as render };
