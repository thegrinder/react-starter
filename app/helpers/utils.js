export const rem = pxValue => `${pxValue / 16}rem`;

export const isEmptyObj = obj => Object.keys(obj).length === 0;

export const isEmptyArr = arr => arr.length === 0;

export const union = (arr1, arr2) => arr1.concat(arr2.filter(i => !arr1.includes(i)));

export const omitBy = (obj, properties) => (
  Object.keys(obj)
    .filter(property => !properties.includes(property))
    .reduce(
      (prev, next) => ({
        ...prev,
        [next]: obj[next],
      }),
      {},
    )
);
