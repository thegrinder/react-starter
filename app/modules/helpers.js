export const initialRequestState = {
  error: {},
  initialLoad: true,
  loading: false,
  succeeded: false,
};

export const createRequest = namespace => {
  const actionTypes = {
    FAILED: `${namespace}/FAILED`,
    FULFILLED: `${namespace}/FULFILLED`,
    LOADING: `${namespace}/LOADING`,
    RESET: `${namespace}/RESET`,
    UPDATE: `${namespace}/UPDATE`,
    SUCCEEDED: `${namespace}/SUCCEEDED`,
    TRIGGER: `${namespace}/TRIGGER`,
  };

  const actionCreators = {
    failed: error => ({ error, type: actionTypes.FAILED }),
    succeeded: () => ({ type: actionTypes.SUCCEEDED }),
    fulfilled: () => ({ type: actionTypes.FULFILLED }),
    loading: () => ({ type: actionTypes.LOADING }),
    reset: () => ({ type: actionTypes.RESET }),
  };

  const reducer = (state = initialRequestState, action) => {
    switch (action.type) {
      case actionTypes.LOADING:
        return {
          ...state,
          error: {},
          loading: true,
          succeeded: false,
        };
      case actionTypes.SUCCEEDED:
        return {
          ...state,
          succeeded: true,
        };
      case actionTypes.FAILED:
        return {
          ...state,
          error: action.error,
        };
      case actionTypes.FULFILLED:
        return {
          ...state,
          initialLoad: false,
          loading: false,
        };
      case actionTypes.RESET:
        return {
          ...initialRequestState,
          initialLoad: state.initialLoad,
        };
      default:
        return state;
    }
  };

  return {
    actionCreators,
    actionTypes,
    reducer,
  };
};
