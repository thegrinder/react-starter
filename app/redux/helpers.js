export const initialRequestState = {
  error: {},
  initialLoad: true,
  loading: false,
};

export const createRequest = (namespace) => {
  const actionTypes = {
    FAILED: `${namespace}/FAILED`,
    FULFILLED: `${namespace}/FULFILLED`,
    LOADING: `${namespace}/LOADING`,
    RESET: `${namespace}/RESET`,
    SUCCEEDED: `${namespace}/SUCCEEDED`,
    TRIGGER: `${namespace}/TRIGGER`,
  };

  const actionCreators = {
    failed: error => ({ error, type: actionTypes.FAILED }),
    fulfilled: () => ({ type: actionTypes.FULFILLED }),
    loading: () => ({ type: actionTypes.LOADING }),
    reset: () => ({ type: actionTypes.RESET }),
  };

  const reducer = (state = initialRequestState, action) => {
    switch (action.type) {
      case actionTypes.LOADING:
        return {
          ...state,
          error: {},
          loading: true,
        };
      case actionTypes.FAILED:
        return {
          ...state,
          error: action.error,
        };
      case actionTypes.FULFILLED:
        return {
          ...state,
          initialLoad: false,
          loading: false,
        };
      case actionTypes.RESET:
        return initialRequestState;
      default:
        return state;
    }
  };

  return {
    actionCreators,
    actionTypes,
    reducer,
  };
};
