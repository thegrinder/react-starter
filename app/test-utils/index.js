/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import PropTypes from 'prop-types';
import configureStore from 'redux-mock-store';
import { IntlProvider } from 'react-intl';
import { ThemeProvider } from 'styled-components';
import * as RTL from '@testing-library/react';

import theme from '../theme';

const propTypes = {
  children: PropTypes.node.isRequired,
};

const Providers = ({ children }) => (
  <IntlProvider locale="en">
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  </IntlProvider>
);

Providers.propTypes = propTypes;

export const render = (ui, options) =>
  RTL.render(ui, { wrapper: Providers, ...options });

export const { fireEvent, act } = RTL;

export const createStore = configureStore();
