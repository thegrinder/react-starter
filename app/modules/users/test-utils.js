import { initialRequestState } from '../helpers';
import { initialUsersState } from './reducer';

export const createUser = (user = {}) => ({
  id: 'id',
  name: 'name',
  username: 'username',
  email: 'email@email.com',
  phone: '132456776',
  ...user,
});

export const createRequestState = (requestState = {}) => ({
  ...initialRequestState,
  ...requestState,
});

export const createUsersState = ({
  requestKey = '',
  requestState = {},
  data = {},
} = {}) => ({
  ...initialUsersState,
  data: {
    ...initialUsersState.data,
    ...data,
  },
  requests: {
    ...initialUsersState.requests,
    ...(requestKey && {
      [requestKey]: {
        ...initialUsersState.requests[requestKey],
        ...requestState,
      },
    }),
  },
});

export const createState = args => ({
  users: createUsersState(args),
});
