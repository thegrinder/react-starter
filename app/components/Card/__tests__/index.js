import React from 'react';

import { render } from '../../../helpers/test-utils';
import Card from '..';


const children = <span>children</span>;

const renderComponent = (props = {}) => render(
  <Card {...props}>{children}</Card>,
);

describe('<Card />', () => {
  it('should render correctly with default props and children', () => {
    const { container: { firstChild }, getByText } = renderComponent();
    const childrenElement = getByText('children');
    expect(firstChild).toBeDefined();
    expect(firstChild).toContainElement(childrenElement);
    expect(firstChild).toMatchSnapshot();
  });

  it('should render <div> tag', () => {
    const { container: { firstChild } } = renderComponent();
    expect(firstChild.tagName).toEqual('DIV');
  });
});
