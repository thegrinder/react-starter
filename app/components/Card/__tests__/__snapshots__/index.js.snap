// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render correctly with default props and children 1`] = `
<div
  class="sc-bdVaJa bcfDey"
>
  <span>
    children
  </span>
</div>
`;
